src/states/common/approved.rs

```diff
@@ -252,7 +257,12 @@ pub trait Approved: Base {
 
     fn chain_poll(&mut self, outbox: &mut dyn EventBox) -> Result<Transition, RoutingError> {
         let mut our_pfx = *self.chain_mut().our_prefix();
-        while let Some((event, neighbour_change)) = self.chain_mut().poll()? {
+        while let Some(AccumulatedEvent {
```

Comment:
would it make sense to not de-structure here, and use `match accumulated_event.event` or is it not working?